daemon/k8s_watcher.go

```diff
@@ -1550,26 +1623,88 @@ func (d *Daemon) addCiliumNetworkPolicyV2(ciliumV2Store cache.Store, cnp *cilium
 		}
 	}
 
-	var cnpns cilium_v2.CiliumNetworkPolicyNodeStatus
 	if err != nil {
-		cnpns = cilium_v2.CiliumNetworkPolicyNodeStatus{
-			OK:          false,
-			Error:       fmt.Sprintf("%s", err),
-			LastUpdated: cilium_v2.NewTimestamp(),
-		}
 		scopedLog.WithError(err).Warn("Unable to add CiliumNetworkPolicy")
 	} else {
-		cnpns = cilium_v2.CiliumNetworkPolicyNodeStatus{
-			OK:          true,
-			LastUpdated: cilium_v2.NewTimestamp(),
-		}
 		scopedLog.Info("Imported CiliumNetworkPolicy")
 	}
 
-	go func() {
-		k8s.UpdateCNPStatusV2(ciliumNPClient.CiliumV2(), ciliumV2Store,
-			k8s.BackOffLoopTimeout, node.GetName(), cnp, cnpns)
-	}()
+	ctrlName := cnp.GetControllerName()
+	cnpCM.UpdateController(ctrlName,
+		controller.ControllerParams{
+			DoFunc: func() error {
+				cnpns := cilium_v2.CiliumNetworkPolicyNodeStatus{
+					OK: true,
+				}
+
+				d.policy.Mutex.RLock()
+				rev := d.policy.GetRevision()
+				d.policy.Mutex.RUnlock()
+
+				cnpns.Revision = rev
```

Comment:
This is not wrong but not ideal. The revision stored here should be the revision returned by `func (p *Repository) Add(r api.Rule) (uint64, error) {` when the CNP was incorporated into the repository.

What you are doing here will cause the revision to be continuously updated. What's interesting to know is what policy revision first implemented the policy.

Reply:
I thought it would be worth it to know if the policy is constantly being enforced, therefore the controller manager. Do you think it's not worth it? 

If yes, how do you suggest to do the first status update for the revision returned in a controller manager while keeping updating the remaining ones?

If no, should we remove the `RunInterval` entirely and let the CM run only for the first update?

Reply:
The controller is fine. I would like the revision being reported to be the first revision that implemented the policy and not the latest revision.

Reply:
But the controller would only exist to do the update for the first revision number, is that the expectation?

Reply:
Yes, the controller is invoked every time a k8s policy is added. Once the policy is added it will be in effect until the policy is removed again. Even when endpoints are being regenerated, the policy will always be in effect.

Reply:
I see, I thought the whole point of the controller would be to keep the most updated revision number of which the policy was being enforced. That would keep the controller running by updating k8s-apiserver with the most recent revision number.

In reality the controller will only be used to keep the revision number from which the policy was initially enforced and keep it in sync with k8s-apiserver until the policy is updated or deleted.

Reply:
Please re-check the code